### ISSUE TITLE
Embedded database support

### ISSUE THREAD

**Issue by sugoidogo** [jc9108/expanse#13]
While browsing this repo I noticed from the [compose.prod.yaml](https://github.com/jc9108/expanse/blob/4f6a94ff1de276805dd5b476bf60f1cbf9857e98/compose.prod.yaml#L34) that expanse is a node.js application who's only dependency outside the NodeJS ecosystem is postgres, a database application. It looks to me like docker is used to run an instance of postgres as a dependency of expanse, but npm also has various [embedded database libraries](https://www.npmjs.com/search?q=keywords%3Aembedded%20database), and there's even [embedded-postgres](https://www.npmjs.com/package/embedded-postgres). Using one of these would allow the removal of docker as a requirement, which opens up many more options for packaging expanse, especially using existing automated tools for nodejs packaging. Using an embedded database by default and allowing the user to [configure an external database at their preference](https://github.com/jc9108/expanse/issues/12) provides a much higher degree of portability to the software that self-hosters like myself would much appreciate.

**Comment by sugoidogo** [comment]
I [tried](https://github.com/jc9108/expanse/commit/330cb338b6ce6a357feba688cb635da36a76e7a0) to implement these changes myself, but I hit a bit of a wall. I added the [dotenv](https://www.npmjs.com/package/dotenv) library and moved the environment variables previously in the launch commands to the `.env_example` to get cross-platform environment variable loading. Since the port was there I also made the bind ip configurable, so hosters can change that to `localhost`/`127.0.0.1` if they only want expanse accessible on their local pc. I gave the connection string the same treatment, adding the `POSTGRES_PORT` variable and a `POSTGRES_EMBEDDED` variable to determine if that info should be used to create the postgres server or not. My issue appeared once I tried to actually use the `embedded-postgres` library.
```
sugoidogo@dev:~/expanse/backend$ npm run prod

> prod
> node -r dotenv/config controller/server.mjs dotenv_config_path=.env_prod

file:///home/sugoidogo/expanse/backend/model/sql.mjs:12
    const pg = new EmbeddedPostgres({
               ^

TypeError: EmbeddedPostgres is not a constructor
    at init_db_embedded (file:///home/sugoidogo/expanse/backend/model/sql.mjs:12:16)
    at Module.init_db (file:///home/sugoidogo/expanse/backend/model/sql.mjs:32:9)
    at file:///home/sugoidogo/expanse/backend/controller/server.mjs:31:11
```
I don't think it's an issue with the library, but there's not much documentation on the [npm package](https://www.npmjs.com/package/embedded-postgres). I don't have a lot of experience with nodejs or successfully using es6 libraries so maybe you can point out what I've done wrong here.

**Comment by sugoidogo** [comment]
Figured out my issue, it's a commonjs module. Have to initialize with `EmbeddedPostgres.default`, don't know why. Tried that before, but assumed it didn't work because the dev misspelled the function as `initialise()` in their code and then again as `inititialize()` on their only doc page. PR coming shortly.

**Comment by jc9108** [comment]
sorry, im not really considering adding this as the app already has docker support and ive resolved #12 earlier to allow using external dbs by providing your own connection string.

if you want to use expanse without docker just run `npm run ci` in both `backend/` and `frontend/`, run `npm run build` in `frontend/`, then run `npm run prod` in `backend/`.

also note that this [embedded-postgres](https://www.npmjs.com/package/embedded-postgres) package is made for testing, it says specifically on their [homepage](https://github.com/zonkyio/embedded-postgres).

so im not going to accept the PR but of course you can keep it in your own fork if it's useful to you.
